sherpa/astro/tests/test_astro_data.py

```diff
@@ -1524,10 +1524,10 @@ def test_notice_channel_grouping_outofbounds_ungrouped(lo, hi, expected, make_da
 @requires_data
 @requires_fits
 @pytest.mark.parametrize("lo,hi,expected",
-                         [(-5, 2000, '0.1248:12.4100'),
+                         [(0, 2000, '0.1248:12.4100'),
                           (0.7, 2000, '0.6716:12.4100'),
-                          (-5, 4.2, '0.1248:4.1391'),
-                          (-20, -5, '0.1248:12.4100'),
+                          (0, 4.2, '0.1248:4.1391'),
+                          (0, 1e-10, '0.1248:12.4100'),
```

Comment:
See https://github.com/sherpa/sherpa/pull/1215#issuecomment-888428804

Reply:
I agree that it comes down to if and how "special" the first call is. I've never understood (and as a consequence always been uncomfortable with) that special first notice call (*). In fact, as a user (before I started reviewing PRs like this) I did not know that that first call was supposed to be special. I noticed that it did not work as I intended and that notice calls did not give repeatable results (i.e. if I repeat the first notice call, I get a different result - except that I never understood that it's only for the **first** call that's different, I just found "notice calls don't give repeatable results). To work around that, my sherpa scripts are littered with statement like `ignore(None, None); notice(None, None)` in an attempt to get to a repeatable state where the following notice/ignore calls give repeatable and predictable results.

So, I strongly dislike the "first notice call is special" design and I will continue to try to work around that in my own analysis. (I find it especially problematic in notebook, where cells can the executed in random order and, if I restart, I don't necessarily run all cells above). However, I understand that that's part of Sherpa's initial design, it's been around for a long time, and we don't want to break user scripts if we can avoid it.

So, I'm accepting that it's special, and at that point don't care how special it is - I'll try to avoid using it as much as I can. (Or how about a "First notice call is special" configuration item?  ;-)  )

(*) I'm sure it's explained somewhere in the docs or threads, but either I never read the right part (most users don't read all thread beginning to end, but pick one that looks close to what they want - I and use it that way, too); or it's a recent addition as want not sufficiently explained when I was a beginning sherpa user.

Reply:
I would argue that we haven't historically documented this well - it's just something you don't really think about too much.

Calling `ignore()` first is one way to get an explicit "clean" state for repeatable calls. But that's outside the scope of this particular PR!